homeassistant/components/vevor_heater/coordinator.py

```diff
@@ -0,0 +1,66 @@
+"""Data update coordinators for Vevor BLE Heater integration."""
+
+import logging
+from typing import Optional
+
+from bleak import BLEDevice
+from vevor_heater_ble.heater import VevorDevice
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_UPDATE_INTERVAL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class VevorHeaterUpdateCoordinator(DataUpdateCoordinator[VevorDevice]):
+    """Update coordinator for Vevor BLE Heaters."""
+
+    def __init__(self, hass: HomeAssistant, device: BLEDevice) -> None:
+        """Initialize the coordinator for the given BLE device. Assumes the device corresponds to a Vevor Heater."""
+        self._ble_device = device
+        self._vevor_device = VevorDevice(
+            address=self._ble_device.address,
+            name="Vevor " + (self._ble_device.name or ""),
```

Comment:
Looks like this isn't using f-strings

Reply:
Different local workflows apparently. I tick off all items I've addressed locally and consider PRs in draft mode as a scratch space which should only be reviewed when explicitly marked as such. Apologies for wasting your time if you're used to a different paradigm. Let me finish this round of comments my way and I'll behave better next time :).

Reply:
Usually I check the review comment against the code before marking it resolved. But it's fine to do it now you are doing it now that I understand your workflow

Reply:
Ack - will be more careful with assumptions next time when contributing to a new project.